_posts/2019-12-31-memkind-dax-kmem.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Memkind support for KMEM DAX option
+author: michalbiesek
+layout: post
+identifier: memkind-support-for-KMEM-DAX-option
+---
+
+### Introduction
+
+Linux kernel version 5.1 brings in [support][kernel-patch] for the volatile-use of persistent memory
+as a hotplugged memory region (**KMEM DAX**).
+From the perspective of OS, persistent memory will be seen as a separate NUMA node(s).
+[libmemkind][memkind-release] extends its API by new kinds which allow for automatic detecting and allocating to
+persistent memory.
+
+### Requirements
+
+At least kernel 5.1 version with enabled **KMEM DAX** option
+
+{% highlight console %}
+$ make nconfig
+	Device Drivers --->
+		-*- DAX: direct access to differentiated memory --->
+			<M>   Device DAX: direct access mapping device
+			<M>     PMEM DAX: direct access to persistent memory
+			<M>     KMEM DAX: volatile-use of persistent memory
+			< >   PMEM DAX: support the deprecated /sys/class/dax interface
+{% endhighlight %}
+
+[ndctl and daxctl][ndctl-release] with version 66 or later.
+
+### System Configuration
+
+Using NVDIMM as an extension of DRAM needs proper configuration with `ndctl` and `daxctl`.
+A Reconfiguration of Device-DAX depends on dax-bus device model.
+Kernel should support `/sys/bus/dax model`. To migrate it from
+`sys/class/dax` to `/sys/bus/dax model` please use [daxctl-migrate-device-mode][daxctl-migrate-device-model].
+
+### Reconfigure dax devices
+
+To see the list of available NUMA nodes on the system you could use `numactl`.
```

Comment:
The list of available NUMA nodes on the system can be retrieved using `numactl`.

Reply:
Done